bundles/core/org.eclipse.smarthome.core.persistence/src/main/java/org/eclipse/smarthome/core/persistence/internal/rest/PersistenceResource.java

```diff
@@ -0,0 +1,237 @@
+            @ApiParam(value = "Start time of the data to return. Will default to 1 day before endtime", required = false) @QueryParam("starttime") String startTime,
+            @ApiParam(value = "End time of the data to return. Will default to current time.", required = false) @QueryParam("endtime") String endTime,
+            @ApiParam(value = "Page number of data to return. This parameter will enable paging.", required = false) @QueryParam("page") int pageNumber,
+            @ApiParam(value = "The length of each page.", required = false) @QueryParam("pagelength") int pageLength) {
+
+        final Object responseObject = getItemHistoryBean(serviceName, itemName, startTime, endTime, pageNumber,
+                pageLength);
+        return Response.ok(responseObject).build();
+    }
+
+    private Date convertTime(String sTime) {
+        DateTimeType dateTime = new DateTimeType(sTime);
+        return dateTime.getCalendar().getTime();
+    }
+
+    private Response getItemHistoryBean(String serviceName, String itemName, String timeBegin, String timeEnd,
+            int pageNumber, int pageLength) {
+        // Benchmarking timer...
+        long timerStart = System.currentTimeMillis();
+
+        // If serviceName is null, then use the default service
+        PersistenceService service = null;
+        if (serviceName == null) {
+        } else {
+            service = persistenceServices.get(serviceName);
+        }
+
+        if (service == null) {
+            logger.debug("Persistence service not found '{}'.", serviceName);
+            return JSONResponse.createErrorResponse(Status.CONFLICT, "Persistence service not found: " + serviceName);
+        }
+
+        if (!(service instanceof QueryablePersistenceService)) {
+            logger.debug("Persistence service not queryable '{}'.", serviceName);
+            return JSONResponse.createErrorResponse(Status.CONFLICT,
+                    "Persistence service not queryable: " + serviceName);
+        }
+
+        QueryablePersistenceService qService = (QueryablePersistenceService) service;
+
+        Date dateTimeBegin = new Date();
+        Date dateTimeEnd = dateTimeBegin;
+        if (timeBegin != null) {
+            dateTimeBegin = convertTime(timeBegin);
+        }
+
+        if (timeEnd != null) {
+            dateTimeEnd = convertTime(timeEnd);
+        }
+
+        // End now...
+        if (dateTimeEnd.getTime() == 0) {
+            dateTimeEnd = new Date();
+        }
+        if (dateTimeBegin.getTime() == 0) {
+            dateTimeBegin = new Date(dateTimeEnd.getTime() - MILLISECONDS_PER_DAY);
+        }
+
+        // Default to 1 days data if the times are the same
+        if (dateTimeBegin.getTime() >= dateTimeEnd.getTime()) {
+            dateTimeBegin = new Date(dateTimeEnd.getTime() - MILLISECONDS_PER_DAY);
+        }
+
+        FilterCriteria filter = new FilterCriteria();
```

Comment:
You can drop this new (or the below) as this object is thrown away without using it.